resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/MountSecretsBootstrap.scala

```diff
@@ -28,20 +28,26 @@ private[spark] class MountSecretsBootstrap(secretNamesToMountPaths: Map[String,
    *
    * @param pod the pod into which the secret volumes are being added.
    * @param container the container into which the secret volumes are being mounted.
+   * @param addNewVolumes whether to add new secret volumes for the secrets.
```

Comment:
I think this problem arose because we're conflating two things here - adding secret volumes (which are pod-scoped) and adding volume-mounts (which are container-scoped). I think we should separate these out. The branching may work for now, but we should have a future work item to separate these out.

cc/ @mccheah

Reply:
Agreed. I didn't separate it out because we will touch this code as part of refactoring the steps code anyway.